ldap_auth_provider.py

```diff
@@ -95,11 +96,14 @@ def check_password(self, user_id, password):
             )
 
             if self.ldap_mode == LDAPMode.SIMPLE:
-                bind_dn = "{prop}={value},{base}".format(
-                    prop=self.ldap_attributes['uid'],
-                    value=localpart,
-                    base=self.ldap_base
-                )
+                if self.ldap_bind_pattern:
+                    bind_dn = self.ldap_bind_pattern.replace("%s", localpart)
+                else:
+                    bind_dn = "{prop}={value},{base}".format(
+                        prop=self.ldap_attributes['uid'],
+                        value=localpart,
+                        base=self.ldap_base
+                    )
```

Comment:
rather than have this conditional here, can we just construct `ldap_bind_pattern` in the config parser:

```python
if 'bind_pattern' not in config:
    ldap_config.bind_pattern = "{prop}=%s,{base}".format(prop=config[attributes["uid"]], base=config["base"])
```